src/java/test/org/apache/zookeeper/ZKTestCase.java

```diff
@@ -75,4 +76,29 @@ public void failed(Throwable e, FrameworkMethod method) {
 
     };
 
+    public interface WaitForCondition {
+        /**
+         * @return true when success
+         */
+        boolean evaluate();
+    }
+
+    /**
+     * Wait for condition to be true; otherwise fail the test if it exceed
+     * timeout
+     * @param msg       error message to print when fail
+     * @param condition condition to evaluate
+     * @param timeout   timeout in seconds
+     * @throws InterruptedException
+     */
+    public void waitFor(String msg, WaitForCondition condition, int timeout)
+            throws InterruptedException {
+        for (int i = 0; i < timeout; ++i) {
+            if (condition.evaluate()) {
+                return;
+            }
+            Thread.sleep(1000);
```

Comment:
I also have such kind of facility in other projects, would it be better to have a inner timeout of 100ms? This may save time